WordPressAuthenticator/Unified Auth/View Related/Get Started/GetStartedViewController.swift

```diff
@@ -34,6 +34,12 @@ public enum SignInError: Error {
     }
 }
 
+/// Error due to site's `xmlrpc.php` file not being public
+///
+public enum XMLRPCError: Error {
+    case xmlrpcError(siteAddress: String)
+}
```

Comment:
I created this error to be able to pass `siteAddress` to the delegate along with this error. I will see whether there is another way to pass that. Kindly let me know if you already know of a way to pass the `siteAddress`.

Reply:
Interesting, I didn't know the reason behind the use of the new type. Maybe we can get the site address on the Woo app by checking the [`currentSelfHostedSite`](https://github.com/woocommerce/woocommerce-ios/blob/7b905f41eb451db04a65a979ba58ffd6e3327113/WooCommerce/Classes/Authentication/AuthenticationManager.swift#L34)? Does it sound too hacky though?

Reply:
Thanks for the input, Huong!

I see that `currentSelfHostedSite` will not be set [if `isSiteDiscovery` is true](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/blob/trunk/WordPressAuthenticator/Unified%20Auth/View%20Related/Site%20Address/SiteAddressViewController.swift#L582-L592). Please correct me if I am missing something.

To avoid dealing with edge cases, I think sticking to `XMLRPCError` is straightforward. What do you think?

Reply:
Another hack: we can also add a new method in the delegate that is similar to the current [`handleError`](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/blob/964b007c21b7c6dbb0760ba5611bd537081ae5c8/WordPressAuthenticator/Authenticator/WordPressAuthenticatorDelegateProtocol.swift#L90) method, but with an extra param for a site address.

Then the default implementation for this method will call the original one; that way it will not affect existing host apps. Does that sound overcomplicated? If it does, we can keep your current solution and iterate later if needed.